stl/inc/memory

```diff
@@ -3064,27 +3064,61 @@ _CXX20_DEPRECATE_OLD_SHARED_PTR_ATOMIC_SUPPORT bool atomic_compare_exchange_stro
 template <class _Ty>
 class alignas(2 * sizeof(void*)) _Atomic_ptr_base {
     // overalignment is to allow potential future use of cmpxchg16b
+
+    static_assert(alignof(_Ref_count_base) >= (1 << 2), "Two bits don't fit as low bits");
+
+    static constexpr uintptr_t _Lock_mask                   = 3;
+    static constexpr uintptr_t _Not_locked                  = 0;
+    static constexpr uintptr_t _Locked_notify_not_needed    = 1;
+    static constexpr uintptr_t _Locked_notify_needed        = 2;
+    static constexpr uintptr_t _Locked_notify_not_supported = 3;
+    static constexpr uintptr_t _Ptr_value_mask              = ~_Lock_mask;
+
 protected:
     constexpr _Atomic_ptr_base() noexcept = default;
 
     _Atomic_ptr_base(_Ty* const _Px, _Ref_count_base* const _Ref) noexcept
         : _Ptr(_Px), _Repptr(reinterpret_cast<uintptr_t>(_Ref)) {}
 
     _NODISCARD _Ref_count_base* _Lock_and_load() const noexcept {
-        constexpr uintptr_t _Low_bit = 1;
-        uintptr_t _Rep               = _Repptr.load(memory_order::relaxed);
+        uintptr_t _Rep = _Repptr.load(memory_order::relaxed);
         for (;;) {
-            _Rep &= ~_Low_bit;
-            if (_Repptr.compare_exchange_weak(_Rep, _Rep | _Low_bit)) {
-                return reinterpret_cast<_Ref_count_base*>(_Rep);
+            switch (_Rep & _Lock_mask) {
+            case _Not_locked: // Can try to lock now
+                if (_Repptr.compare_exchange_weak(_Rep, _Rep | _Locked_notify_not_needed)) {
+                    return reinterpret_cast<_Ref_count_base*>(_Rep);
+                }
+                _YIELD_PROCESSOR();
+                break;
+
+            case _Locked_notify_not_supported: // Keep spinning
+                _YIELD_PROCESSOR();

```

Comment:
This should almost certainly be something like `SwitchToThread()`

Reply:
Actually it looks like this bit is never set so this case should probably be removed.

Reply:
It was an attempt to preserve compatibility with old ABI, if old ABI is also changed. Now this branch should be like `default: abort();` and `_Locked_notify_not_supported` should be removed.

Reply:
Removed.
(As it is now clarified that headers or import libraries don't need to support XP, as well as implementation source files, I hope we don't need `SwitchToThread` waiting ever, we have at the very least SRW Lock)